distributed/worker.py

```diff
@@ -1691,9 +1699,26 @@ def transition_ready_executing(self, ts):
                 pdb.set_trace()
             raise
 
+    def transition_ready_error(self, ts):
+        if self.validate:
+            assert ts.exception is not None
+            assert ts.traceback is not None
+        self.send_task_state_to_scheduler(ts)
+
     def transition_ready_memory(self, ts, value=None):
+        if value:
+            self.put_key_in_memory(ts, value=value)
         self.send_task_state_to_scheduler(ts)
 
+    def transition_ready_waiting(self, ts):
+        """
+        This transition is common for work stealing
+        """
+        ts.state = "waiting"
+        ts.runspec = None
+
+        return ts.state
```

Comment:
Do we need to transition to waiting in this case?  I'm curious if this was necessary before due to the transition when we were releasing keys, but maybe it's not necessary now?

Reply:
Also, in both cases where we call this the final state is "waiting".  Do we need to set the state explicitly, or can we let the `transition` function handle this?

Reply:
This transition is probably not necessary. The only case I am aware of where we hit this is work stealing.